File: src/de/schafunschaf/bountiesexpanded/scripts/campaign/intel/bounties/RareFlagshipData.java
package de.schafunschaf.bountiesexpanded.scripts.campaign.intel.bounties;

import lombok.Getter;

import java.util.Set;

@Getter
public class RareFlagshipData {
    private final String flagshipID;
    private final String flagshipVariantID;
    private final Set<String> factionIDs;
    private final float weight;

    public RareFlagshipData(String flagshipID, String flagshipVariantID, Set<String> factionIDs, float weight) {
        this.flagshipID = flagshipID;
        this.flagshipVariantID = flagshipVariantID;
        this.factionIDs = factionIDs;
        this.weight = weight;
    }
}

File: src/de/schafunschaf/bountiesexpanded/data/ships/RareFlagshipData.kt
package de.schafunschaf.bountiesexpanded.data.ships

val RARE_FLAGSHIPS = HashMap<String, RareFlagshipData>()

class RareFlagshipData(
    private var flagshipID: String,
    private var flagshipVariantID: String,
    private var factionIDs: Set<String>,
    private var weight: Float
)